packages/nx/src/generators/tree.ts

```diff
@@ -240,11 +248,31 @@ export class FsTree implements Tree {
     }
   }
 
-  children(dirPath: string): string[] {
+  children(dirPath: string): string[];
+  children(dirPath: string, recursive: boolean);
+  children(dirPath: string, recursive?: boolean) {
     dirPath = this.normalize(dirPath);
-    let res = this.fsReadDir(dirPath);
+    let res: string[] = [];
+
+    if (recursive) {
+      const readDir = (dir: string) => {
+        let files = this.fsReadDir(dir);
+        files = [...files, ...this.directChildrenOfDir(this.rp(dir))];
+        files.forEach((file) => {
+          const fileAbsolutePath = join(dir, file);
```

Comment:
Could you help me to track changes on the tree instead of the file system please. Because of it, some tests are failing.